File: src/main/java/com/mbayou/kick4k/events/EventsClient.java
package com.mbayou.kick4k.events;

import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.mbayou.kick4k.KickConfiguration;
import com.mbayou.kick4k.api.ApiClient;
import com.mbayou.kick4k.authorization.AuthorizationClient;

import java.net.http.HttpClient;
import java.util.List;
import java.util.Map;

public class EventsClient extends ApiClient {
    public EventsClient(HttpClient httpClient, ObjectMapper mapper, KickConfiguration configuration, AuthorizationClient authorization) {
        super(httpClient, mapper, configuration, authorization);
    }

    public List<EventSubscription> getEventsSubscriptions() {
        return this.get(this.configuration.getEvents())
                .send(new TypeReference<>() {});
    }

    public List<EventSubscriptionResponse> postEventsSubscription(EventSubscriptionRequest request) {
        return this.post(this.configuration.getEvents())
                .body(request)
                .send(new TypeReference<>() {});
    }

    public void deleteEventsSubscriptions(String... ids) {
        this.delete(this.configuration.getEvents())
                .queryParams(Map.of("id", ids))
                .send(new TypeReference<>() {});
    }

    public void deleteEventsSubscriptions(List<String> ids) {
        this.delete(this.configuration.getEvents())
                .queryParams(Map.of("id", ids))
                .send(new TypeReference<>() {});
    }
}

File: src/main/kotlin/com/mbayou/kick4k/events/EventsClient.kt
package com.mbayou.kick4k.events

import com.fasterxml.jackson.databind.ObjectMapper
import com.mbayou.kick4k.KickConfiguration
import com.mbayou.kick4k.api.ApiClient
import java.net.http.HttpClient

class EventsClient(
    httpClient: HttpClient,
    mapper: ObjectMapper,
    configuration: KickConfiguration,
): ApiClient(httpClient, mapper, configuration) {

    fun getEventsSubscriptions(accessToken: String): List<EventSubscription> {
        return get(configuration.events)
            .withAccessToken(accessToken)
            .send(responseType<List<EventSubscription>>()) ?: emptyList()
    }

    fun postEventsSubscription(accessToken: String, request: EventSubscriptionRequest): List<EventSubscriptionResponse> {
        return post(configuration.events)
            .body(request)
            .withAccessToken(accessToken)
            .send(responseType<List<EventSubscriptionResponse>>()) ?: emptyList()
    }

    fun deleteEventsSubscriptions(accessToken: String, vararg ids: String) {
        delete(configuration.events)
            .queryParams(mapOf("id" to ids))
            .withAccessToken(accessToken)
            .send(responseType<Unit>())
    }

    fun deleteEventsSubscriptions(accessToken: String, ids: List<String>) {
        delete(configuration.events)
            .queryParams(mapOf("id" to ids))
            .withAccessToken(accessToken)
            .send(responseType<Unit>())
    }
}
